pkg/reconcilermanager/controllers/reposync_controller_test.go

```diff
@@ -261,19 +264,31 @@ func rolebinding(name, reconcilerName string, opts ...core.MetaMutator) *rbacv1.
 	return result
 }
 
-func setupNSReconciler(t *testing.T, objs ...client.Object) (*syncerFake.Client, *RepoSyncReconciler) {
+type repoReconcilerTest struct {
```

Comment:
Until I found this impl, I wad _realy_ confused at why you'd replaced the client with the reconciler variable...

This makes it a lot more confusing to know what methods are being called on the client or the dynamic client or the reconciler itself...

Why not just have setupNSReconciler return 3 variables? 

This isn't java. Go supports multiple return values. You don't need to wrap everything in a class to return it.

Reply:
The reason I decided to wrap these three things is because the 3 variables are not always needed for all the times we call this function, sometime we only need 1 or 2,  the code changes to be like _,_,XX = setupNSReconciler() or XX,_XX = ..., makes me feel it is better to wrap the 3 variables.  And we use these 3 variables for our reconciler test, so I think it makes sense to put them together.